anime-archive/src/components/carousel/Carousel.js

```diff
@@ -19,12 +24,42 @@ const Carousel = (props) => {
     newFilter.appendChild(dot);
   };
 
+  // Holds data for carousel cards and will change based on filter selected in carousel
+  const [carouselCardData, setCarouselCardData] = useState(null);
+
+  // Loader
+  const [carouselLoading, setCarouselLoading] = useState(false);
+
   // Grabs first filter in data structure and make it the default filter
   const defaultFilter = filterData[0];
 
   // Removes the default filter so we wont display it twice
   const filters = filterData.slice(1);
 
+  // Tracks filter change and holds newly selected query sorting name
+  const [queryFilter, setQueryFilter] = useState("POPULARITY_DESC");
+
+  // Dynamic filter call to api triggered on everytime a new is selected
+  useEffect(() => {
+    setCarouselLoading(true);
+    axios
+      .post("https://graphql.anilist.co", {
+        query: fetchCarousel,
+        variables: { filterName: queryFilter },
+      })
+      .then(function (response) {
+        setCarouselCardData(response);
+
+        setTimeout(() => {
```

Comment:
Let's make it faster :), I'd say remove the set timeout to make our app more snappy. Also, animation and new styling looks so  CLEAN! Great job.

Reply:
Just pushed a commit with setTimeouts removed from api calls

Reply:
Awesome, super fast and clean! Lets Gooooooo!